stable/yugabyte/templates/service.yaml

```diff
@@ -255,46 +255,35 @@ spec:
         # Set the anti-affinity selector scope to YB masters.
         {{ if $root.Values.AZ }}
         nodeAffinity:
-          requiredDuringSchedulingIgnoredDuringExecution:
-            nodeSelectorTerms:
-            - matchExpressions:
-              - key: failure-domain.beta.kubernetes.io/zone
-                operator: In
-                values:
-                - {{ $root.Values.AZ }}
-            - matchExpressions:
-              - key: topology.kubernetes.io/zone
-                operator: In
-                values:
-                - {{ $root.Values.AZ }}
+          {{- $baseAffinity := include "yugabyte.multiAZNodeAffinity" $root | fromYaml -}}
+          {{- if eq .name "yb-masters" -}}
+          {{- $mergedAffinity := get $root.Values.master.affinity "nodeAffinity" | default (dict) | mustMergeOverwrite $baseAffinity -}}
+          {{ toYaml $mergedAffinity | nindent 10}}
+          {{- else -}}
+          {{- $mergedAffinity := get $root.Values.tserver.affinity "nodeAffinity" | default (dict)  | mustMergeOverwrite $baseAffinity -}}
+          {{ toYaml $mergedAffinity | nindent 10}}
+          {{- end -}}
         {{ end }}
         podAntiAffinity:
-          preferredDuringSchedulingIgnoredDuringExecution:
-          - weight: 100
-            podAffinityTerm:
-              labelSelector:
-                matchExpressions:
-                {{- if $root.Values.oldNamingStyle }}
-                - key: app
-                  operator: In
-                  values:
-                  - "{{ .label }}"
-                {{- else }}
-                - key: app.kubernetes.io/name
-                  operator: In
-                  values:
-                  - "{{ .label }}"
-                - key: release
-                  operator: In
-                  values:
-                  - {{ $root.Release.Name | quote }}
-                {{- end }}
-              topologyKey: kubernetes.io/hostname
+          {{- $basePodAntiAffinity := include "yugabyte.podAntiAffinity" ($appLabelArgs) | fromYaml -}}
```

Comment:
Is it possible to do the merge in the dict form and then render this together with the affinity section on 277.